cli/pkg/workspace/plugin/kubeconfig.go

```diff
@@ -132,187 +134,189 @@ func (o *UseWorkspaceOptions) Run(ctx context.Context) error {
-			if !found {
-				return fmt.Errorf("cluster %q not found in kubeconfig", currentContext.Cluster)
-			}
-			prev = prev.DeepCopy()
-			prev.Cluster = kcpCurrentWorkspaceContextKey
-			newKubeConfig.Clusters[kcpCurrentWorkspaceContextKey] = prevCluster
-		}
-		newKubeConfig.Contexts[kcpCurrentWorkspaceContextKey] = prev
-		newKubeConfig.Contexts[kcpPreviousWorkspaceContextKey] = currentContext
-
-		newKubeConfig.CurrentContext = kcpCurrentWorkspaceContextKey
-
-		if err := o.modifyConfig(o.ClientConfig.ConfigAccess(), newKubeConfig); err != nil {
-			return err
-		}
-
-		newServerHost = newKubeConfig.Clusters[newKubeConfig.Contexts[kcpCurrentWorkspaceContextKey].Cluster].Server
-
-		bindings, err := o.getAPIBindings(ctx, o.kcpClusterClient, newServerHost)
-		if err != nil {
-			// display the error, but don't stop the current workspace from being reported.
-			fmt.Fprintf(o.ErrOut, "error checking APIBindings: %v\n", err)
-		}
-		if err = findUnresolvedPermissionClaims(o.Out, bindings); err != nil {
-			// display the error, but don't stop the current workspace from being reported.
-			fmt.Fprintf(o.ErrOut, "error checking APIBindings: %v\n", err)
-		}
-
-		return currentWorkspace(o.Out, newServerHost, shortWorkspaceOutput(o.ShortWorkspaceOutput), nil)
+	// LEGACY(mjudeikis): Remove once everybody gets used to this
+	if strings.HasPrefix(o.Name, "root") {
+		o.Name = ":" + o.Name
+		fmt.Fprintf(o.Out, "Note: Using 'root:' to define an absolute path is no longer supported. Instead, use ':root' to specify an absolute path.\n")
+	}
 
-	case "..":
-		config, err := o.ClientConfig.ClientConfig()
-		if err != nil {
-			return err
-		}
-		u, currentClusterName, err := pluginhelpers.ParseClusterURL(config.Host)
-		if err != nil {
-			return fmt.Errorf("current URL %q does not point to a workspace", config.Host)
-		}
-		parentClusterName, hasParent := currentClusterName.Parent()
-		if !hasParent {
-			if currentClusterName == core.RootCluster.Path() {
-				return fmt.Errorf("current workspace is %q", currentClusterName)
-			}
-			return fmt.Errorf("current workspace %q has no parent", currentClusterName)
-		}
-		u.Path = path.Join(u.Path, parentClusterName.RequestPath())
-		newServerHost = u.String()
+	// validate early for valid name
+	err = o.validate(ctx)
+	if err != nil {
+		return err
+	}
 
-	case "":
+	switch {
+	// o.Name empty and home workspace should be always one after another for fallthrough to
+	// print error and default to home.
+	case o.Name == "":
 		defer func() {
 			if err == nil {
 				_, err = fmt.Fprintf(o.Out, "Note: 'kubectl ws' now matches 'cd' semantics: go to home workspace. 'kubectl ws -' to go back. 'kubectl ws .' to print current workspace.\n")
 			}
 		}()
 		fallthrough
 
-	case "~", home:
-		homeWorkspace, err := o.kcpClusterClient.Cluster(core.RootCluster.Path()).TenancyV1alpha1().Workspaces().Get(ctx, "~", metav1.GetOptions{})
+	case o.Name == "~" || o.Name == home:
```

Comment:
what about `~:foo` ?

Reply:
added support, just no test as need second run on it :/
~ not being resolved in tests to check if workspace exists gets bit complex in tests :./